tests/test_cupy.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from array_api_compat import device, to_device
+
+xp = pytest.importorskip("array_api_compat.cupy")
+from cupy.cuda import Stream
+
+
+def test_to_device_with_stream():
+    devices = xp.__array_namespace_info__().devices()
+    streams = [
+        Stream(),
+        Stream(non_blocking=True), 
+        Stream(null=True),
+        Stream(ptds=True), 
+        123,  # lapack stream
```

Comment:
```suggestion
        123,  # dlpack stream
```

typo? unlikely that linear algebra is anyhow related